modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/options/JavaOptions.java

```diff
@@ -0,0 +1,295 @@
+package org.openapitools.codegen.languages.options;
+
+import org.apache.commons.lang3.StringEscapeUtils;
+import org.openapitools.codegen.CodegenConfig;
+import org.openapitools.codegen.DefaultCodegen;
+import org.openapitools.codegen.CodegenConstants;
+import org.openapitools.codegen.languages.AbstractJavaCodegen;
+
+import java.util.*;
+
+import static org.openapitools.codegen.languages.AbstractJavaCodegen.FULL_JAVA_UTIL;
+
+public class JavaOptions extends DefaultOptions implements Options {
+
+    public JavaOptions() {
+        super();
+
+        supportsInheritance = true;
+
+        setReservedWordsLowerCase(
+                Arrays.asList(
+                        // used as internal variables, can collide with parameter names
+                        "localVarPath", "localVarQueryParams", "localVarCollectionQueryParams",
+                        "localVarHeaderParams", "localVarFormParams", "localVarPostBody",
+                        "localVarAccepts", "localVarAccept", "localVarContentTypes",
+                        "localVarContentType", "localVarAuthNames", "localReturnType",
+                        "ApiClient", "ApiException", "ApiResponse", "Configuration", "StringUtil",
+
+                        // language reserved words
+                        "abstract", "continue", "for", "new", "switch", "assert",
+                        "default", "if", "package", "synchronized", "boolean", "do", "goto", "private",
+                        "this", "break", "double", "implements", "protected", "throw", "byte", "else",
+                        "import", "public", "throws", "case", "enum", "instanceof", "return", "transient",
+                        "catch", "extends", "int", "short", "try", "char", "final", "interface", "static",
+                        "void", "class", "finally", "long", "strictfp", "volatile", "const", "float",
+                        "native", "super", "while", "null")
+        );
+
+        languageSpecificPrimitives = new HashSet<String>(
+                Arrays.asList(
+                        "String",
+                        "boolean",
+                        "Boolean",
+                        "Double",
+                        "Integer",
+                        "Long",
+                        "Float",
+                        "Object",
+                        "byte[]")
+        );
+
+        typeMapping.put("date", "Date");
+        typeMapping.put("file", "File");
+
+        instantiationTypes.put("array", "ArrayList");
+        instantiationTypes.put("map", "HashMap");
+    }
+
+    @Override
+    public void processOpts(CodegenConfig codegen) {
+        super.processOpts(codegen);
+        this.doProcessOpts((AbstractJavaCodegen) codegen);
+    }
+
+    public void doProcessOpts(AbstractJavaCodegen codegen) {
```

Comment:
I think this would eventually not make a lot of sense.

Once language-specifics are separated from generator types, having an "Abstract" base type specific to a language wouldn't be too meaningful. The idea is that this would eventually enable us to break `--generator-name` options into three options:

* framework
* language
* type

(see #137)

This would allow users to target templates modified for their target language and reuse any framework-specific logic. As an example, if someone wanted to create a NancyFX generator using F# (which has different reserved words and types than C#), they could target the nancyfx specific code and some new F# language options.

Looking at the implementation, though, I'm wondering how we'd differentiate settings this options in a meaningful way from the generators.

Maybe @jmini might have some input since he's familiar with the discussion in the issue linked above?

Reply:
You're right. I tried to keep backward compatibility and not erasing the AbstractJavaCodegen seemed easiest for that. Considering the comment [above](https://github.com/OpenAPITools/openapi-generator/pull/852#discussion_r212175674) I can see that extracting the processOpts is not right.